lib/instrumentation/async-hooks.js

```diff
@@ -1,641 +1,37 @@
-          ensureObWrapper(promise, false)
-          return resolve(val)
-        }
-
-        function wrappedReject (val) {
-          ensureObWrapper(promise, false)
-          return reject(val)
-        }
-      }
-    }
-
-    util.inherits(wrappedPromise, Promise)
-
-    wrap(Promise.prototype, 'then', wrapThen)
-    // Node.js <v7 only, alias for .then
-    if (Promise.prototype.chain) {
-      wrap(Promise.prototype, 'chain', wrapThen)
-    }
-
-    var PromiseFunctions = [
-      'all',
-      'race',
-      'reject',
-      'resolve',
-      'accept',  // Node.js <v7 only
-      'defer'    // Node.js <v7 only
-    ]
-
-    PromiseFunctions.forEach(function (key) {
-      // don't break `in` by creating a key for undefined entries
-      if (typeof Promise[key] === 'function') {
-        wrappedPromise[key] = Promise[key]
-      }
-    })
-
-    global.Promise = wrappedPromise
-
-    function ensureObWrapper (promise, overwrite) {
-      if (!promise.__ob_wrapper || overwrite) {
-        promise.__ob_wrapper = ins.bindFunction(propagateObWrapper)
-      }
-    }
-
-    function propagateObWrapper (ctx, fn, result, next) {
-      var nextResult
-      try {
-        nextResult = fn.call(ctx, result)
-        return {returnVal: nextResult, error: false}
-      } catch (err) {
-        return {errorVal: err, error: true}
-      } finally {
-        // Wrap any resulting futures as continuations.
-        if (nextResult instanceof Promise) {
-          next.__ob_wrapper = function proxyWrapper () {
-            var obWrapper = nextResult.__ob_wrapper || propagateObWrapper
-            return obWrapper.apply(this, arguments)
-          }
-        } else {
-          ensureObWrapper(next, true)
-        }
-      }
-    }
-
-    function wrapThen (original) {
-      return function wrappedThen () {
-        var promise = this
-        var next = original.apply(promise, Array.prototype.map.call(arguments, bind))
-
-        next.__ob_wrapper = function proxyWrapper (ctx, fn, val, last) {
-          if (promise.__ob_wrapper) {
-            promise.__ob_wrapper(ctx, function () {}, null, next)
-            return next.__ob_wrapper(ctx, fn, val, last)
-          }
-          return propagateObWrapper(ctx, fn, val, last)
-        }
-
-        return next
-
-        // wrap callbacks (success, error) so that the callbacks will be called as a
-        // continuations of the resolve or reject call using the __ob_wrapper created above.
-        function bind (fn) {
-          if (typeof fn !== 'function') return fn
-          return ins.bindFunction(function (val) {
-            var result = (promise.__ob_wrapper || propagateObWrapper)(this, fn, val, next)
-            if (result.error) {
-              throw result.errorVal
-            } else {
-              return result.returnVal
-            }
-          })
-        }
-      }
-    }
+  function before (asyncId) {
+    if (!initState.has(asyncId)) return // in case type === TIMERWRAP
```

Comment:
We need to consider if the benefit of filtering out `TIMERWRAP` types outweigh the overhead of these `has()` calls